Cannot build latest release from source

Steps to reproduce:

- Download the Terraform source code.
- Check out Git tag v1.1.5.
- Run `GO111MODULE=on go install -v ./...` using the latest Go release (go1.17.7).

I see the following error:

```
$ go install -v ./...
go: downloading github.com/hashicorp/go-tfe v0.19.1-0.20211020175229-e52963e079d0
internal/backend/remote/backend.go:15:2: github.com/hashicorp/go-tfe@v0.19.1-0.20211020175229-e52963e079d0: invalid version: unknown revision e52963e079d0
```

If I look in the Github repository for hashicorp/go-tfe, I can't find a tag for commit v0.19.1 or a commit SHA for e52963e079d0.

I'm able to resolve the issue by updating go-tfe to v0.24.0.

--- a/go.mod
+++ b/go.mod
@@ -42,7 +42,7 @@ require (
 	github.com/hashicorp/go-plugin v1.4.3
 	github.com/hashicorp/go-retryablehttp v0.7.0
 	github.com/hashicorp/go-safetemp v1.0.0
-	github.com/hashicorp/go-tfe v0.19.1-0.20211020175229-e52963e079d0
+	github.com/hashicorp/go-tfe v0.21.0
 	github.com/hashicorp/go-uuid v1.0.2
 	github.com/hashicorp/go-version v1.3.0
 	github.com/hashicorp/hcl v0.0.0-20170504190234-a4b07c25de5f
@@ -180,7 +180,7 @@ require (
 	go.uber.org/multierr v1.1.0 // indirect
 	go.uber.org/zap v1.10.0 // indirect
 	golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5 // indirect
-	golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba // indirect
+	golang.org/x/time v0.0.0-20211116232009-f0f3c7e86c11 // indirect
 	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
 	google.golang.org/genproto v0.0.0-20210319143718-93e7006c17a6 // indirect

--- a/internal/cloud/backend_apply_test.go
+++ b/internal/cloud/backend_apply_test.go
@@ -13,6 +13,7 @@ import (
 	gomock "github.com/golang/mock/gomock"
 	"github.com/google/go-cmp/cmp"
 	tfe "github.com/hashicorp/go-tfe"
+	mocks "github.com/hashicorp/go-tfe/mocks"
 	version "github.com/hashicorp/go-version"
 	"github.com/hashicorp/terraform/internal/addrs"
 	"github.com/hashicorp/terraform/internal/backend"
@@ -576,7 +577,7 @@ func TestCloud_applyAutoApprove(t *testing.T) {
 	defer bCleanup()
 	ctrl := gomock.NewController(t)
 
-	applyMock := tfe.NewMockApplies(ctrl)
+	applyMock := mocks.NewMockApplies(ctrl)
 	// This needs three new lines because we check for a minimum of three lines
 	// in the parsing of logs in `opApply` function.
 	logs := strings.NewReader(applySuccessOneResourceAdded)
@@ -778,7 +779,7 @@ func TestCloud_applyWithAutoApprove(t *testing.T) {
 	defer bCleanup()
 	ctrl := gomock.NewController(t)
 
-	applyMock := tfe.NewMockApplies(ctrl)
+	applyMock := mocks.NewMockApplies(ctrl)
 	// This needs three new lines because we check for a minimum of three lines
 	// in the parsing of logs in `opApply` function.
 	logs := strings.NewReader(applySuccessOneResourceAdded)
@@ -1269,7 +1270,7 @@ func TestCloud_applyPolicySoftFailAutoApproveSuccess(t *testing.T) {
 	defer bCleanup()
 	ctrl := gomock.NewController(t)
 
-	policyCheckMock := tfe.NewMockPolicyChecks(ctrl)
+	policyCheckMock := mocks.NewMockPolicyChecks(ctrl)
 	// This needs three new lines because we check for a minimum of three lines
 	// in the parsing of logs in `opApply` function.
 	logs := strings.NewReader(fmt.Sprintf("%s\n%s", sentinelSoftFail, applySuccessOneResourceAdded))
@@ -1289,7 +1290,7 @@ func TestCloud_applyPolicySoftFailAutoApproveSuccess(t *testing.T) {
 	policyCheckMock.EXPECT().Logs(gomock.Any(), gomock.Any()).Return(logs, nil)
 	policyCheckMock.EXPECT().Override(gomock.Any(), gomock.Any()).Return(nil, nil)
 	b.client.PolicyChecks = policyCheckMock
-	applyMock := tfe.NewMockApplies(ctrl)
+	applyMock := mocks.NewMockApplies(ctrl)
 	// This needs three new lines because we check for a minimum of three lines
 	// in the parsing of logs in `opApply` function.
 	logs = strings.NewReader("\n\n\n1 added, 0 changed, 0 destroyed")
@@ -1348,7 +1349,7 @@ func TestCloud_applyPolicySoftFailAutoApprove(t *testing.T) {
 	defer bCleanup()
 	ctrl := gomock.NewController(t)
 
-	applyMock := tfe.NewMockApplies(ctrl)
+	applyMock := mocks.NewMockApplies(ctrl)
 	// This needs three new lines because we check for a minimum of three lines
 	// in the parsing of logs in `opApply` function.
 	logs := strings.NewReader(applySuccessOneResourceAdded)

